src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/Common/CommonTypes.Windows.cs

```diff
@@ -31,31 +34,263 @@ public interface IInspectableInterface { }
+        public static readonly Guid IID_IUnknown = new("00000000-0000-0000-c000-000000000046");
+        public static readonly Guid IID_IDispatch = new("00020400-0000-0000-c000-000000000046");
+        public static readonly Guid IID_IClassFactory = new("00000001-0000-0000-c000-000000000046");
+
+        public const int S_OK = 0;
+        public const int E_NOTIMPL = unchecked((int)0x80004001);
+        public const int E_NOINTERFACE = unchecked((int)0x80004002);
+    }
+
+    /// <summary>
+    /// Class factory used to provide a unmanaged IClassFactory instance for testing with <see cref="ComObject"/>.
+    /// </summary>
+    internal unsafe struct ComObjectFactory
+    {
+        public const string CLSID = "3DCAAC37-F484-49E9-9257-273BA9618162";
+
+        [ModuleInitializer]
+        internal static unsafe void RegisterInProcCOMServer()
+        {
+            var clsid = new Guid(ComObjectFactory.CLSID);
+            const int CLSCTX_INPROC_SERVER = 1;
+            const int REGCLS_MULTIPLEUSE = 1;
+            void* classFactory = ComObjectFactory.Create();
+            int res = CoRegisterClassObject(in clsid, classFactory, CLSCTX_INPROC_SERVER, REGCLS_MULTIPLEUSE, out int cookie);
+            Xunit.Assert.Equal(ComConstants.S_OK, res);
+            Marshal.Release((IntPtr)classFactory);
+
+            [DllImport("Ole32")]
+            static extern int CoRegisterClassObject(in Guid clsid, void* factory, int clsContext, int flags, out int registerCookie);
+        }
+
+        // Create an instance of a COM object class factory
+        public static void* Create()
+        {
+            var vtable = (VTable*)NativeMemory.Alloc((nuint)sizeof(VTable));
+            vtable->QueryInterface = &QueryInterface;
+            vtable->AddRef = &AddRef;
+            vtable->Release = &Release;
+
+            vtable->CreateInstance = &CreateInstance;
+            vtable->LockServer = &LockServer;
+
+            var instance = (ComObjectFactory*)NativeMemory.Alloc((nuint)sizeof(ComObjectFactory));
+            instance->_vtable = vtable;
+            instance->_refCount = 1;
+            return instance;
+        }
+
+        // The COM ABI for a vtable
+        private struct VTable
+        {
+            // IUnknown
+            public delegate* unmanaged[Stdcall]<void*, Guid*, void**, int> QueryInterface;
+            public delegate* unmanaged[Stdcall]<void*, uint> AddRef;
+            public delegate* unmanaged[Stdcall]<void*, uint> Release;
+
+            // IClassFactory
+            public delegate* unmanaged[Stdcall]<void*, void*, Guid*, void**, int> CreateInstance;
+            public delegate* unmanaged[Stdcall]<void*, int, int> LockServer;
+        }
+
+        // The COM ABI requires the first pointer field to be the vtable
+        private VTable* _vtable;
+
+        // Additional instance fields for this COM object
+        private uint _refCount;
+
+        [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvStdcall) })]
```

Comment:
The explicit Stdcall everywhere is unnecessary. (And it would be actually wrong in the hypothetical case this was ever needed on non-Windows.)

Reply:
For built-in COM types, I've been explicit about this, but your observation is fair. Would you like me to remove them?

Reply:
I think it would look better without it. Less verbose code is better in general.